onnxruntime/contrib_ops/cpu/bert/multihead_attention_helper.h

```diff
@@ -167,6 +170,7 @@ Status CheckInputs(const T* query,
             "Expect 'key' shape (batch_size, num_heads, kv_sequence_length, head_size) for past_key");
       }
 
+      qkv_format = QKV_BSN3H; // key is past_key, so qkv must be packed
```

Comment:
@wangyems, K and V are BNSH, What's Q format? Maybe set qkv_format=UNKNOWN here.

Reply:
Then it is a new qkv format like Q_K_V_BSNH_BNSH_BNSH.

@wangyems, it is a little bit complex, maybe we shall use Q_K_V_BNSH instead (Q is also BNSH).

Reply:
I think you guys should not relay on the tensor rank and various shape configuration to infer/derive the qkv format, instead, do it in reverse, add an attribute `qkv_format`, and `ORT_ENFORCE` the shapes with it. Then the code logic will be crystal clear. You will also not be bothered by the ambiguous shape config in the future, in case you will introduce more packing logic.

Reply:
The original design of qkv_format is to indicate the format of Q/K/V after AddBiasTranspose.

For this case, I suggest to mark as UNKOWN to indicate that it is not supported yet in Rocm EP.

Reply:
@wangyems @tianleiwu, do I need to do something? Otherwise, I think it is good to go and merge it.

Reply:
qkv_format = UNKNOWN